File: cli/src/commands/boards.ts
import { requireApiKey } from '../config.js'
import { api } from '../api.js'
import { accentBold, dim, board } from '../theme.js'

interface Board {
  id: string
  label: string
  postCount: number
}

export async function boardsCommand(): Promise<void> {
  const key = requireApiKey()
  const res = await api<Board[]>('/boards', key)
  const boards = res.data ?? []

  if (boards.length === 0) {
    console.log(dim('No boards found.'))
    return
  }

  console.log()
  console.log(accentBold('Boards'))
  console.log()

  const maxLabel = Math.max(...boards.map((b) => b.label.length))
  for (const b of boards) {
    const label = b.label.padEnd(maxLabel + 2)
    const count = dim(`${b.postCount} posts`)
    console.log(`  ${board(b.id).padEnd(maxLabel + 12)} ${label} ${count}`)
  }
  console.log()
}

File: cli-rust/src/commands/boards.rs
use anyhow::Result;
use serde::Deserialize;

use crate::api::ArkoraApi;
use crate::config;
use crate::theme;

#[derive(Deserialize)]
#[serde(rename_all = "camelCase")]
struct Board {
    id: String,
    label: String,
    post_count: i64,
}

pub async fn run(color: (u8, u8, u8)) -> Result<()> {
    let cfg = config::load();
    let key = config::require_key(&cfg);
    let api = ArkoraApi::new(&config::api_url(&cfg), &key);

    let resp = api.get::<Vec<Board>>("/boards").await?;
    let boards = resp.data.unwrap_or_default();

    if boards.is_empty() {
        println!("{}", theme::dim("No boards found."));
        return Ok(());
    }

    println!();
    println!("{}", theme::accent_bold("Boards", color));
    println!();

    for b in &boards {
        println!(
            "  {:<20}  {:<20}  {}",
            theme::board(&b.id, color),
            b.label,
            theme::dim(&format!("{} posts", b.post_count))
        );
    }
    println!();

    Ok(())
}
